### PR TITLE
Bluetooth-test and Message structure improvements

### PR DESCRIPTION
**Pull request by theJenix**
These commits get us back to bluetooth-test functionality, with some additional functionality (e.g. Let Me In/FindNewFans button).  This implements the new discovery style (though not the whole new discovery process), and supports discovering multiple fans at once.  Tested using my Nexus4 and Reid's test Galaxy Nexus...there is some bad connection latency when there are a lot of other non-fanClub bluetooth devices nearby, and we need some way of detecting when a socket is dropped, but otherwise it tested fine.

### CODE REVIEW

File: fanClub/src/com/lastcrusade/fanclub/AcceptThread.java [hunk 1]
```diff
@@ -3,54 +3,38 @@
...
         BluetoothServerSocket tmp = null;
         try {
             // MY_UUID is the app's UUID string, also used by the client code
-            tmp = mBluetoothAdapter.listenUsingRfcommWithServiceRecord(
+            //TODO: for scaling to large number of users, we may have to use different UUIDs
+            mmServerSocket = adapter.listenUsingRfcommWithServiceRecord(
                     HOST_NAME,
                     UUID.fromString(context.getString(R.string.app_uuid)));
         } catch (IOException e) {
-        }
```
Comment (reidbaker) [inline fanClub/src/com/lastcrusade/fanclub/AcceptThread.java:29]: If there is an IO exception what does that mean. Lets put a comment saying why it isn't important or let's log something.

File: fanClub/src/com/lastcrusade/fanclub/ConnectThread.java [hunk 2]
```diff
@@ -7,47 +7,50 @@
...
-                }
-            } catch (IOException closeException) {
-            }
-            return null;
+            Log.e(TAG, "Unable to connect socket", e);
+            cancel();
         }
 
-        return socket;
+        return mmSocket;
```
Comment (reidbaker) [inline fanClub/src/com/lastcrusade/fanclub/ConnectThread.java:53]: What is the mm stand for? I know its convention but I was curious.

Comment (theJenix) [inline fanClub/src/com/lastcrusade/fanclub/ConnectThread.java:53]: It's for member variables.  Android example code uses it, and I/we copied that in from the Bluetooth walkthrough page.  We can change the convention to confirm with our other code, if desired.

File: fanClub/src/com/lastcrusade/fanclub/HostActivity.java [hunk 3]
```diff
@@ -144,6 +186,11 @@ protected void onDeviceFound(BluetoothAdapter adapter, Intent intent) {
             }
         }
         try {
+            //one thread per device found...if there are multiple devices,
+            // there are multiple threads
+            //TODO: asyncTask may not work....if the host discovers 4 devices, it appears to still only use 1 async task thread
+            // and if the first 3 of those devices are not fanclub, itll pause for a while attempting to conenct, which will delay
+            // connection of the actual fan
```
Comment (reidbaker) [inline fanClub/src/com/lastcrusade/fanclub/HostActivity.java:193]: nitpick, lets make the spacing to the right of `//` the same

File: fanClub/src/com/lastcrusade/fanclub/util/BluetoothUtils.java [hunk 4]
```diff
@@ -5,26 +5,50 @@
...
             if (!adapter.isEnabled()) {
-                adapter.enable();
-                if (!adapter.isEnabled()) {
+                Toaster.iToast(context, R.string.enable_bluetooth);
+                boolean enableStarted = adapter.enable();
+                boolean enabled = false;
+                if (enableStarted) {
+                    //once the enable process is started, it takes some time to be enabled
+                    int retrySeconds = 5;
+                    int retryPauseMS = 500;
```
Comment (reidbaker) [inline fanClub/src/com/lastcrusade/fanclub/util/BluetoothUtils.java]: Is MS milliseconds? Either way let comments say the units

File: fanClub/src/com/lastcrusade/fanclub/util/BluetoothUtils.java [hunk 5]
```diff
@@ -5,26 +5,50 @@
...
         if (adapter != null) {
             if (!adapter.isEnabled()) {
-                adapter.enable();
-                if (!adapter.isEnabled()) {
+                Toaster.iToast(context, R.string.enable_bluetooth);
+                boolean enableStarted = adapter.enable();
+                boolean enabled = false;
+                if (enableStarted) {
+                    //once the enable process is started, it takes some time to be enabled
+                    int retrySeconds = 5;
```
Comment (reidbaker) [inline fanClub/src/com/lastcrusade/fanclub/util/BluetoothUtils.java]: Was this determined by trial and error?

File: fanClub/src/com/lastcrusade/fanclub/HostActivity.java (hunk 3 above)
Comment (reidbaker) [inline fanClub/src/com/lastcrusade/fanclub/HostActivity.java:193]: Didnt realize the spaces were marking a single thought while trying to keep lines short. 
These are fine.

File: fanClub/src/com/lastcrusade/fanclub/util/BluetoothUtils.java (hunk 5 above)
Comment (theJenix) [inline fanClub/src/com/lastcrusade/fanclub/util/BluetoothUtils.java]: To some degree...I observed that it took some time to enable the adapter, and it never took greater than 5 seconds (not anywhere close).  I'll add a comment to the code.

File: fanClub/src/com/lastcrusade/fanclub/AcceptThread.java (hunk 1 above)
Comment (theJenix) [inline fanClub/src/com/lastcrusade/fanclub/AcceptThread.java:29]: It throws an exception...see the code on line 37 below.

File: fanClub/src/com/lastcrusade/fanclub/util/BluetoothUtils.java (hunk 5 above)
Comment (theJenix) [inline fanClub/src/com/lastcrusade/fanclub/util/BluetoothUtils.java]: comment added to the code explaining the numbers.  I also moved the values out into constants.

**Pull request merged**